src/wallet/partialTxProposal.ts

```diff
@@ -0,0 +1,319 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { PartialTx, PartialTxInputData } from '../models/partial_tx';
+import Address from '../models/address';
+import P2SH from '../models/p2sh';
+import P2PKH from '../models/p2pkh';
+import Transaction from '../models/transaction';
+import { AddressError, InvalidPartialTxError, InsufficientFundsError } from '../errors';
+import wallet from '../wallet';
+import Network from '../models/network';
+import { HATHOR_TOKEN_CONFIG } from '../constants';
+
+import transaction from '../transaction';
+import helpers from '../utils/helpers';
+
+import { OutputType } from './types';
+
+class PartialTxProposal {
+
+  network: Network;
+  public partialTx: PartialTx;
+  public signatures: PartialTxInputData|null;
+  private transaction: Transaction|null;
+
+  /**
+   * @param {Network} network
+   */
+  constructor(network: Network) {
+    this.network = network;
+    this.partialTx = new PartialTx(this.network);
+    this.signatures = null;
+    this.transaction = null;
+  }
+
+  /**
+   * Create a PartialTxProposal instance from the serialized string.
+   *
+   * @param {string} serialized Serialized PartialTx data
+   * @param {Network} network network
+   *
+   * @throws {SyntaxError} serialized argument should be a valid PartialTx.
+   * @throws {UnsupportedScriptError} All outputs should be P2SH or P2PKH
+   *
+   * @returns {PartialTxProposal}
+   */
+  static fromPartialTx(serialized: string, network: Network) {
+    const partialTx = PartialTx.deserialize(serialized, network);
+    const proposal = new PartialTxProposal(network);
+    proposal.partialTx = partialTx;
+    return proposal;
+  }
+
+  /**
+   * Add inputs sending the amount of tokens specified, may add a change output.
+   *
+   * @param {string} token UID of token that is being sent
+   * @param {number} value Quantity of tokens being sent
+   * @param {Object} [options]
+   * @param {string|null} [options.changeAddress=null] If we add change, use this address instead of getting a new one from the wallet.
+   * @param {boolean} [options.markAsSelected=true] Mark the utxo with `selected_as_input`.
+   *
+   * @throws InsufficientFundsError
+   */
+  addSend(
+    token: string,
+    value: number,
+    { changeAddress = null, markAsSelected = true }: { changeAddress?: number|null, markAsSelected?: boolean } = {},
+  ) {
+    this.resetSignatures();
+
+    const walletData = wallet.getWalletData();
```

Comment:
I'm thinking about this method. With this, you require the user to run one of the methods we have in `src/wallet.js` to load the wallet in the local storage, right? Maybe this class here could receive a `HathorWallet` objects, and this method here could use the `getUtxos` method. What do you think? It makes much more sense to me thinking about stop using the old methods from the lib.